### ISSUE TITLE
Rename validateEmail options to align with detectBot and sensitiveInfo options

### ISSUE THREAD

**Issue by blaine-arcjet** [arcjet/arcjet-js#1834]
The `validateEmail` currently uses `block` to indicate which email types to deny. We should change this to `deny` like `detectBot` and `sensitiveInfo` now use.

Also, should we add an `allow` variant here? cc @davidmytton

**Comment by davidmytton** [comment]
Yes I think we should match the API of the others, including `allow`.
